hypothesis-python/src/hypothesis/core.py

```diff
@@ -176,7 +176,12 @@ def decode_failure(blob):
     if prefix == b'\0':
         return buffer[1:]
     elif prefix == b'\1':
-        return zlib.decompress(buffer[1:])
+        try:
+            return zlib.decompress(buffer[1:])
+        except zlib.error:
+            raise InvalidArgument(
+                'Invalid zlib compression for blob %r' % (blob,)
```

Comment:
How big could be blob here? Wouldn't it blow up stderr?

Reply:
The absolute maximum would be 8MB run through `b64encode`, or whatever the user has set as the `buffer_size` setting.  In practice, 3K would be an unusually large buffer with no shrinking, and typical failure messages would have a buffer of at most a few hundred characters.

So the test is definitely failing and the output won't be pretty, but the traceback would usually be longer than the buffer and stderr OK regardless.

Reply:
Hm...ok, let's hope it wouldn't make boom one day (: